src/librustc_resolve/diagnostics.rs

```diff
@@ -68,6 +71,16 @@ crate fn add_typo_suggestion(
         err.span_suggestion(
             span, &msg, suggestion.candidate.to_string(), Applicability::MaybeIncorrect
         );
+        let def_span = suggestion.res.opt_def_id()
+            .map(|def_id| resolver.definitions.opt_span(def_id)
+                .unwrap_or_else(|| resolver.cstore.get_span_untracked(def_id, resolver.session)));
```

Comment:
This should be something like
```rust
match def_id.krate {
    LOCAL_CRATE => resolver.definitions.opt_span(def_id),
    _ => Some(resolver.definitions.opt_span(def_id)),
}
```
instead of `unwrap_or_else` because if the def-id is local, but `definitions` for some reason doesn't have a span for it (e.g. the span is dummy), then we'll go into `cstore` with a local crate ID and get an ICE.